### PR TITLE
Lockfreeonly locations

### PR DESCRIPTION
**Pull request by Dashy-Dolphin**

### CODE REVIEW

File: src/kcas.ml [hunk 1]
```diff
@@ -174,8 +174,8 @@ let rec update_no_alloc backoff loc state set_after =
     let backoff = Backoff.once backoff in
     update_no_alloc backoff loc state set_after
 
-let is_obstruction_free casn =
-  Atomic.get casn == (Mode.obstruction_free :> status)
+let is_obstruction_free casn loc =
+  Atomic.get casn == (Mode.obstruction_free :> status) && loc.lock_mode == Mode.lock_free
```
Comment (polytypic) [inline src/kcas.ml:178]: Should the comparison be against `Mode.obstruction_free`?

File: src/kcas.ml [hunk 2]
```diff
@@ -189,8 +189,8 @@ let cas loc before state =
 module Loc = struct
   type 'a t = 'a loc
 
-  let make after =
-    { state = Atomic.make @@ new_state after; id = Id.get_unique () }
+  let make ?(mode_val = Mode.lock_free) after =
```
Comment (polytypic) [inline src/kcas.ml:192]: I think `Mode.obstruction_free` should be the default.

Comment (polytypic) [reply src/kcas.ml:192]: Also, the label could be just `mode`, i.e. `?(mode = Mode.obstruction_free)`.

Comment (Dashy-Dolphin) [reply src/kcas.ml:192]: Will modify them accordingly.

File: src/kcas.ml (hunk 1 above)
Comment (Dashy-Dolphin) [reply src/kcas.ml]: Yes , It should be against Mode.obstruction_free.😅

**Pull request closed without merging**